### PR TITLE
Doc 11647 prometheus discovery

### PR DESCRIPTION
**Pull request by ggray-cb**
Added documentation for Prometheus discovery API to 7.2 docs. Changed pages, with links to the preview site:

- [Added "New in 7,2,1" subsection](https://preview.docs-test.couchbase.com/prometheus-updates-7.2/server/current/introduction/whats-new.html#new-in-version-7-2-1) to What's New and included a description of the new feature.
- [Configure Prometheus to Collect Couchbase Metrics](https://preview.docs-test.couchbase.com/prometheus-updates-7.2/server/current/manage/monitor/set-up-prometheus-for-monitoring.html) rewritten to cover new API.
- [Added new REST API](https://preview.docs-test.couchbase.com/prometheus-updates-7.2/server/current/rest-api/rest-discovery-api.html) page for the discovery endpoint.

### CODE REVIEW

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc [hunk 1]
```diff
@@ -1,107 +1,117 @@
 = Set up Prometheus to Consume Couchbase Metrics
 
-:description: This page will demonstrate how to enable a Prometheus server to read from Couchbase's Prometheus-ready endpoints.
+:description: Couchbase Server provides a API endpoint that helps you configure Prometheus to collect data from it.
```
Comment (stevewatanabe) [inline modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: Should this be "an API" ?

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc [hunk 2]
```diff
@@ -1,107 +1,117 @@
...
 
-Make sure that you have https://www.docker.com/get-started[Docker] installed and running on your machine. 
-You should also have an available Couchbase cluster with at least one running node.
+For a general introduction to using Prometheus with Couchbase, see the blog post https://www.couchbase.com/blog/scraping-database-metrics-from-couchbase-capella-with-prometheus/[Scraping Database Metrics from Couchbase Capella with Prometheus^].
 
-:sectnums:
+To collect metrics from your Couchbase Server database, Prometheus must connected to the nodes in its cluster. Therefore, it needs to have a list of the cluster's nodes. To make this process easier and help automate updating Prometheus's list of nodes, Couchbase Server provides a discovery API endpoint named `/prometheus_sd_config`. Calling this endpoint returns a list of the nodes in the cluster in a format that Prometheus can use.
```
Comment (stevewatanabe) [inline modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: "must connected" -> "must connect"

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc [hunk 3]
```diff
@@ -1,107 +1,117 @@
...
-[source, console]
-----
-wget --content-disposition \
-    'http://10.112.211.101:8091/prometheus_sd_config.yaml' \
-    --http-user Administrator --http-password password
-----
+== Prerequisites
 
-This will generate a configuration file containing the target addresses of the nodes in your cluster. 
+You should create a Couchbase Server user for Prometheus to use when connecting to your database. Assign this user the External State Reader role which grants it the ability to call the discovery API to be able to read Couchbase Server's metrics.
```
Comment (stevewatanabe) [inline modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: "External State" -> "External Stats"

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc [hunk 4]
```diff
@@ -1,107 +1,117 @@
...
-== Install Prometheus
+With file-based configuration, you directly call the API discovery endpoint using `curl` or `wget` and save the output to a file. 
+By default, the endpoint returns the list of all nodes in the cluster in JSON format. 
+Prometheus accepts the list in either JSON or YAML format. 
 
-Open a terminal window and run the following command to provision Prometheus:
+To use the file-based method:
 
+. Call the Discovery API endpoint `/prometheus_sd_config` and save the results to a file. 
+The following example calls the discovery API endpoint using the `prometheus` user on a Couchbase node named `node1` and saves the output to a file named `couchbase_nodes.json`:
```
Comment (stevewatanabe) [inline modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: "node1" -> "node01" (to match line 63 below: http://node01:8091/prometheus_sd_config)

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc [hunk 5]
```diff
@@ -1,107 +1,117 @@
...
-    prom/prometheus
+curl  -u prometheus:_password_ \
+      http://node01:8091/prometheus_sd_config \
+      > couchbase_nodes.json
 ----
-<.> The port through which the Prometheus server is accessed and takes the form: `-p{nbsp}host-port:container-port`.
+. Copy the file to your Prometheus host if you did not run `curl` on your Prometheus host. 
+. Once the files are in place on the Prometheus host, edit the `prometheus.yml` configuration file. 
+Add a new job to collect metrics from your Couchbase Database. 
+In this job, add a `file_sd_configs` key entry that points to the list of nodes you created earlier.
```
Comment (stevewatanabe) [inline modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: "points to the list of nodes" -> "points to the file containing the list of nodes returned via the call to /prometheus_sd_config".

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc (hunk 4 above)
Comment (stevewatanabe) [reply modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: "node1" is used in the example for http discovery so this line should remain "node1" and "node01" in line 63 should be "node1".

File: modules/rest-api/pages/rest-discovery-api.adoc [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+When set to `external`, it uses the node's xref:learn:clusters-and-availability/connectivity.adoc#alternate-addresses[alternate address]. 
+
+port=[insecure|secure]::
+Controls which port Couchbase Server uses in the list of nodes. 
+When set to the default `secure`, it uses the node's secure port in the list.
+When set to `insecure`, it uses the node's unencrypted port.
+
+type=[json|yaml]::
+Controls the data format Couchbase Server uses for the node list. 
+When set ot the default `json`, it returns the list of nodes in JSON format.
```
Comment (stevewatanabe) [inline modules/rest-api/pages/rest-discovery-api.adoc]: "ot" -> "to"

File: modules/rest-api/pages/rest-discovery-api.adoc [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+When set to `yaml`, it returns the list of nodes in YAML format. 
+
+== Examples
+
+The following example demonstrates calling the `prometheus_sd_config` endpoint on a node with the hostname `node01` using default values.
```
Comment (stevewatanabe) [inline modules/rest-api/pages/rest-discovery-api.adoc]: "node01" -> "node1"

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc (hunk 1 above)
Comment (ggray-cb) [reply modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: Yup. Fixed.

File: modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc (hunk 5 above)
Comment (ggray-cb) [reply modules/manage/pages/monitor/set-up-prometheus-for-monitoring.adoc]: Good point. Reworded to state the value needs to be the path to the file.

**Pull request merged**